yarpl/src/yarpl/observable/Subscriptions.cpp

```diff
@@ -12,12 +13,20 @@ namespace observable {
  */
 void Subscription::cancel() {
   cancelled_ = true;
+  for(auto& subscription : *tiedSubscriptions_.rlock()) {
+    subscription->cancel();
+  }
 }
 
 bool Subscription::isCancelled() const {
   return cancelled_;
 }
 
+void Subscription::tieSubscription(Reference<Subscription> subscription) {
+  CHECK(subscription);
+  tiedSubscriptions_->push_back(std::move(subscription));
```

Comment:
What should happen if you call `tieSubscription` after `cancel`? This seems racey.

Reply:
What would you like to see happening? I would suggest that after the subscription is canceled, any tied subscriptions are ignored.

Reply:
Because it's now nondeterministic if a subscription that you tie into a subscription that's about to be canceled will ever receive a cancel. There's no paired `onSubscribe` to signal if a subscription was added before a cancel (and if the Subscription *will* have its cancel called), so the application developer won't know if they're throwing a `Subscription` into the ether or not.

Reply:
Agree. I will change it so that is the case the original subscription is cancelled, I will cancel all tied subscriptions right away.